src/fsharp/Optimizer.fs

```diff
@@ -3180,6 +3187,13 @@ and OptimizeBinding cenv isRec env (TBind(vref, expr, spBind)) =
         let exprOptimized, einfo = 
             let env = if vref.IsCompilerGenerated && Option.isSome env.latestBoundId then env else {env with latestBoundId=Some vref.Id} 
             let cenv = if vref.InlineInfo = ValInline.PseudoVal then { cenv with optimizing=false} else cenv 
+            let cenv = 
+                // Do not optimize an expression that simply wraps another lambda with an application of a value.
```

Comment:
Might need a better comment. Something like:
```Do not optimize an expression that wraps a function call for a bound value. On application of a value, it will inline the function anyway and possibly inline the inner lambda if it can. This is to prevent unwanted function allocations had the function performed a simple inline on application.```